huaweicloud/services/dms/resource_huaweicloud_dms_rabbitmq_plugin.go

```diff
@@ -0,0 +1,320 @@
+		return nil, retry, err
+	}
+	_, err = common.RetryContextWithWaitForState(&common.RetryContextWithWaitForStateParam{
+		Ctx:          ctx,
+		RetryFunc:    retryFunc,
+		WaitFunc:     rabbitmqInstanceStateRefreshFunc(deleteRabbitmqPluginClient, instanceID),
+		WaitTarget:   []string{"RUNNING"},
+		Timeout:      d.Timeout(schema.TimeoutDelete),
+		DelayTimeout: 1 * time.Second,
+		PollInterval: 10 * time.Second,
+	})
+	if err != nil {
+		return diag.Errorf("error disabling the plugin: %s", err)
+	}
+
+	// The RabbitMQ disabling plugin is done if the status of its task is SUCCESS.
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"CREATED"},
+		Target:       []string{"SUCCESS"},
+		Refresh:      rabbitmqPluginTaskRefreshFunc(deleteRabbitmqPluginClient, instanceID, d, "rabbitmqPluginModify"),
+		Timeout:      d.Timeout(schema.TimeoutDelete),
+		Delay:        1 * time.Second,
+		PollInterval: 2 * time.Second,
+	}
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.Errorf("error waiting for the RabbitMQ (%s) disabling plugin (%s) to be done: %s", instanceID, d.Get("name"), err)
+	}
+
+	return resourceDmsRabbitmqPluginRead(ctx, d, cfg)
+}
+
+func rabbitmqPluginTaskRefreshFunc(client *golangsdk.ServiceClient, instanceID string, d *schema.ResourceData,
+	taskName string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		// getRabbitmqPluginTask: get RabbitMQ plugin task
+		getRabbitmqPluginTaskHttpUrl := "v2/{project_id}/instances/{instance_id}/tasks"
+		getRabbitmqPluginTaskPath := client.Endpoint + getRabbitmqPluginTaskHttpUrl
+		getRabbitmqPluginTaskPath = strings.ReplaceAll(getRabbitmqPluginTaskPath, "{project_id}",
+			client.ProjectID)
+		getRabbitmqPluginTaskPath = strings.ReplaceAll(getRabbitmqPluginTaskPath, "{instance_id}", instanceID)
+
+		reqOpt := golangsdk.RequestOpts{
+			KeepResponseBody: true,
+		}
+		getRabbitmqPluginTaskResp, err := client.Request("GET", getRabbitmqPluginTaskPath, &reqOpt)
+
+		if err != nil {
+			return nil, "QUERY ERROR", err
+		}
+
+		getRabbitmqPluginTaskRespBody, err := utils.FlattenResponse(getRabbitmqPluginTaskResp)
+		if err != nil {
+			return nil, "PARSE ERROR", err
+		}
+
+		task := getRabbitmqPluginTask(taskName, d, getRabbitmqPluginTaskRespBody)
+		if task == nil {
+			return nil, "NIL ERROR", fmt.Errorf("can not find the task")
+		}
+		status := utils.PathSearch("status", task, "").(string)
+		return task, status, nil
+	}
+}
+
+func getRabbitmqPluginTask(taskName string, d *schema.ResourceData, resp interface{}) interface{} {
+	rawName := d.Get("name")
+	pluginTasks := utils.PathSearch(fmt.Sprintf("tasks|[?name=='%s']", taskName), resp, make([]interface{}, 0)).([]interface{})
+
+	for _, task := range pluginTasks {
+		params := utils.PathSearch("params", task, nil).(string)
+		paramsData := []byte(params)
+		var paramsJons interface{}
+		err := json.Unmarshal(paramsData, &paramsJons)
+		if err != nil {
+			fmt.Println(err)
```

Comment:
should it return nil?
btw, we shoud use log.Printf instead of fmt.Println

Reply:
yes, it's possible to delete the task. Replace fmt.Println with log.Printf.